extendeddatacrossword.go

```diff
@@ -6,32 +6,30 @@ import (
 	"fmt"
 )
 
+// axis represents which of a row or col.
+type Axis int
+
 const (
-	row = iota
-	col
+	Row Axis = iota
+	Col
 )
 
 // ErrUnrepairableDataSquare is thrown when there is insufficient chunks to repair the square.
 var ErrUnrepairableDataSquare = errors.New("failed to solve data square")
 
-// ErrByzantineRow is thrown when a repaired row does not match the expected row Merkle root.
-type ErrByzantineRow struct {
-	RowNumber uint     // Row index
-	Shares    [][]byte // Pre-repaired row shares. Missing shares are nil.
-}
-
-func (e *ErrByzantineRow) Error() string {
-	return fmt.Sprintf("byzantine row: %d", e.RowNumber)
+// ErrByzantineData is thrown when a repaired row or column does not match the expected row or column Merkle root.
+type ErrByzantineData struct {
+	Axis   Axis     // Axis of the data.
+	Index  uint     // Row/Col index.
+	Shares [][]byte // Pre-repaired shares. Missing shares are nil.
 }
 
-// ErrByzantineCol is thrown when a repaired column does not match the expected column Merkle root.
-type ErrByzantineCol struct {
-	ColNumber uint     // Column index
-	Shares    [][]byte // Pre-repaired column shares. Missing shares are nil.
-}
-
-func (e *ErrByzantineCol) Error() string {
-	return fmt.Sprintf("byzantine column: %d", e.ColNumber)
+func (e *ErrByzantineData) Error() string {
+	if e.Axis == Row {
+		return fmt.Sprintf("byzantine row: %d", e.Index)
+	} else {
+		return fmt.Sprintf("byzantine column: %d", e.Index)
+	}
```

Comment:
Switch?

Reply:
Fixed in [778a54a](https://github.com/celestiaorg/rsmt2d/pull/80/commits/778a54a263a7612dfe51ab66a527b586ab774b86)